### PR TITLE
Delete unused columns codereview and builder_version from files.File model. (bug 1171273)

### PR DESCRIPTION
**Pull request by berkerpeksag**
https://bugzilla.mozilla.org/show_bug.cgi?id=1171273

### CODE REVIEW

File: apps/files/models.py [hunk 1]
```diff
@@ -163,7 +158,6 @@ def from_upload(cls, upload, version, platform, is_beta=False,
             file_.jetpack_version = data['sdkVersion'][:10]
         if file_.jetpack_version:
             Tag(tag_text='jetpack').save_tag(addon)
-        file_.builder_version = data['builderVersion']
```
Comment (kmaglione) [inline apps/files/models.py]: This needs to be removed from `get_jetpack_metadata` as well.

File: apps/files/tasks.py [hunk 2]
```diff
@@ -248,11 +248,6 @@ def start_upgrade(file_ids, sdk_version=None, priority='low', **kw):
                 'secret': settings.BUILDER_SECRET_KEY,
                 'uuid': data['uuid'],
                 'pingback': absolutify(reverse('amo.builder-pingback'))}
-        if file_.builder_version:
-            post['package_key'] = file_.builder_version
-        else:
-            # Older jetpacks might not have builderVersion in their harness.
-            post['location'] = file_.get_url_path('builder')
```
Comment (kmaglione) [inline apps/files/tasks.py]: This whole function can be removed now.

File: apps/files/tests/test_tasks.py [hunk 3]
```diff
@@ -42,33 +42,6 @@ def file(self, **file_values):
...
-
-    def test_jetpack_with_older_harness_opt(self):
-        file_ = self.file(builder_version=None)  # old harness options
-        tasks.start_upgrade([file_.id], sdk_version='1.2')
-        url, args = self.urllib2.urlopen.call_args[0]
-        args = dict(urlparse.parse_qsl(args))
-        eq_(args['location'], file_.get_url_path('builder'))
-        assert 'package_key' not in args, (
-            'Unexpected keys: %s' % args.keys())
-
```
Comment (kmaglione) [inline apps/files/tests/test_tasks.py]: All of the repackager code and tests should really be removed, but I can't think of a good reason to remove these two specifically.

File: apps/files/utils.py [hunk 4]
```diff
@@ -530,8 +530,6 @@ def find_jetpacks(minver, maxver, from_builder_only=False):
                                  version__addon__disabled_by_user=False)
              .exclude(status=amo.STATUS_DISABLED).no_cache()
              .select_related('version'))
-    if from_builder_only:
-        files = files.exclude(builder_version=None)
```
Comment (kmaglione) [inline apps/files/utils.py]: Please remove the argument too.

**Comment by kmaglione** [comment]
You have some failing flake8 tests for unused imports: https://travis-ci.org/mozilla/olympia/jobs/75761887

**Comment by kmaglione** [comment]
Looks good, thanks. Just some minor issues to address. r+wc

**Comment by wagnerand** [comment]
Don't we need a db migration script to actually drop the tables from the db?

**Comment by berkerpeksag** [comment]
> Don't we need a db migration script to actually remove the columns from the db table?

Yes, there is already one that does that :) Please see `migrations/826-files-delete-unused-columns.sql`.

File: apps/files/models.py (hunk 1 above)
Comment (berkerpeksag) [inline apps/files/models.py]: Done.

File: apps/files/tasks.py (hunk 2 above)
Comment (berkerpeksag) [inline apps/files/tasks.py]: Done.

File: apps/files/utils.py (hunk 4 above)
Comment (berkerpeksag) [inline apps/files/utils.py]: Done.

File: apps/files/tests/test_tasks.py (hunk 3 above)
Comment (berkerpeksag) [inline apps/files/tests/test_tasks.py]: I probably wasn't very careful :) Also removed the repackage_jetpack function in tasks.py.

**Comment by berkerpeksag** [comment]
I've addresses all review comments in https://github.com/berkerpeksag/olympia/commit/a1a736b8aece911e51b79d2292f35f1a9fcdd516. I didn't squash the two commits to make review easier.

Thanks for the review!

File: apps/addons/fixtures/addons/featured.json [hunk 5]
```diff
@@ -263,14 +262,12 @@
...
         "fields": {
             "status": 4,
-            "codereview": 0,
             "hash": "sha256:5b5a",
             "created": "2007-03-05 13:19:15",
             "modified": "2007-04-04 12:30:11",
             "filename": "del.icio.us_bookmarks-1.0.43-fx.xpi",
             "platform": 1,
             "version": 1003,
-            "modified": "2007-04-04 12:30:11",
```
Comment (magopian) [inline apps/addons/fixtures/addons/featured.json]: Why remove this line?

Comment (berkerpeksag) [inline apps/addons/fixtures/addons/featured.json]: It's a duplicate of line 267:

```
             "modified": "2007-04-04 12:30:11",
```

See https://github.com/mozilla/olympia/pull/661/files#diff-11ddc2768fbda00caaf8145fa453fb25R267

Comment (magopian) [inline apps/addons/fixtures/addons/featured.json]: ah, you're right, well spotted ;)

**Comment by magopian** [comment]
That's a big change, thanks. It's missing the removal of the `Addon.is_packaged` field that's also mentioned in the [bug 1171273](https://bugzilla.mozilla.org/show_bug.cgi?id=1171273) right?

There's the `BUILDER_VERSIONS_URL` setting that we should be able to remove from the various settings files, and its usage in `zadmin/forms.py` (maybe simply drop the `JetpackUpgradeForm`?), and the associated tests in `zadmin/tests/test_views.py` (`test_no_builder_versions` and `test_get_builder_versions`).

There's also still some test failures because of bad imports.

**Comment by berkerpeksag** [comment]
If I read bug 1171273 correctly, the is_packaged column in the files table is a leftover from https://github.com/mozilla/olympia/blob/master/migrations/462-move-is-packaged.sql#L6 and it probably only exist on prod and/or staging databases. Perhaps I can update the migration script to delete that column, if exists?

Sorry about the import errors, will fix them today. Thanks!

**Comment by magopian** [comment]
Ah, I thought it was talking about the `Addon.is_packaged` field (which doesn't seems to be used either anymore).

@kmaglione what do you think, should this field also be removed? Should we have a migration to remove the column on the File?

**Comment by kmaglione** [comment]
Let's leave the `Addon.is_packaged` field for now. I have a partial patch to remove all of the app fields from the `Addon` model. I just got hung up when I tried to remove the related front-end code, and got to the point where I couldn't tell what was needed and what wasn't. I need to rip out the front end changes, then I'll submit it some time this week.

**Comment by berkerpeksag** [comment]
I've now addressed all review comments. Thanks.

**Comment by magopian** [comment]
important note: the migration now needs to be renamed to 827, since a PR with migration 826 was merged lately.

File: apps/files/tasks.py [hunk 6]
```diff
@@ -58,155 +40,6 @@ def extract_file(viewer, **kw):
...
-        send_upgrade_email(addon, new_version, file_data['version'])
-    except Exception:
-        jp_log.error(msg('Could not send success email.'), exc_info=True)
-        raise
-
-    # Return the new file to make testing easier.
-    return new_file
-
-
 def send_upgrade_email(addon, new_version, sdk_version):
```
Comment (magopian) [inline apps/files/tasks.py]: It seems that this function isn't used anymore, so I think it can go to.

File: lib/settings_base.py [hunk 7]
```diff
@@ -1250,8 +1250,6 @@ def read_only_mode(env):
 BUILDER_SECRET_KEY = 'love will tear us apart'
 # The builder URL we hit to upgrade jetpacks.
 BUILDER_UPGRADE_URL = 'https://addons.mozilla.org/services/builder'
```
Comment (magopian) [inline lib/settings_base.py]: I believe we can get completely rid of all those `BUILDER_*` settings, right? @kmaglione?

File: migrations/826-files-delete-unused-columns.sql [hunk 8]
```diff
@@ -0,0 +1,3 @@
+ALTER TABLE `files`
```
Comment (magopian) [inline migrations/826-files-delete-unused-columns.sql]: This file needs to be renamed to `827-files-delete-unused-columns.sql` now that there's a `826...` migration on master.

**Comment by magopian** [comment]
@berkerpeksag congratz on pulling this off, it's huge! A few last remaining nitpicks, and we'll be ready for merge ;)

File: apps/files/tasks.py (hunk 6 above)
Comment (berkerpeksag) [inline apps/files/tasks.py]: Done.

File: lib/settings_base.py (hunk 7 above)
Comment (berkerpeksag) [inline lib/settings_base.py]: Done.

File: migrations/826-files-delete-unused-columns.sql (hunk 8 above)
Comment (berkerpeksag) [inline migrations/826-files-delete-unused-columns.sql]: Done.

**Comment by berkerpeksag** [comment]
Rebased and pushed. Thanks for the quick reviews!

**Comment by magopian** [comment]
Excellent! Thanks for this huge work!

r+

**Pull request merged**